src/afrolabs/components/kafka.clj

```diff
@@ -1546,12 +1574,20 @@
                             (.get)
                             (set))
 
+        desired-topics (->> (mapcat #(get-topic-names %) topic-name-providers)
+                            (distinct))
+
+        ;; make sure topics have the correct number of partitions
+        _ (when increase-existing-partitions?
+            (increase-topic-partitions! @ac
+                                        (filter existing-topics desired-topics)
+                                        nr-of-partitions))
```

Comment:
side-effect in the middle of the let binding, to try and keep related things together. Also, first increase partitions then change compaction strategy... Not sure which order is better here...